devito/compiler.py

```diff
@@ -290,14 +272,30 @@ def load(basename, compiler=GNUCompiler):
     return npct.load_library(lib_file, '.')
 
 
-def jit_compile_and_load(ccode, basename, compiler=GNUCompiler):
-    """JIT compile the given ccode
+def jit_compile(ccode, compiler=GNUCompiler):
+    """JIT compile the given ccode.
 
     :param ccode: String of C source code.
-    :param basename: The string used to name various files for this compilation.
     :param compiler: The toolchain used for compilation. GNUCompiler by default.
-    :return: The compiled library.
+
+    :return: The name of the compilation unit.
     """
-    jit_compile(ccode, basename, compiler=compiler)
 
-    return load(basename, compiler=compiler)
+    hash_key = sha1(str(ccode).encode()).hexdigest()
+    basename = path.join(get_tmp_dir(), hash_key)
```

Comment:
I wanted ``compiler.py`` to have the freedom to decide how to encode things like the filename etc. 

But yes, you're probably right about caching. Perhaps we can simply keep track of the returned basename in the Operator itself?